hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/ReflectiveCompactSerializer.java

```diff
@@ -167,7 +167,8 @@ private static List<Field> getAllFields(List<Field> fields, Class<?> type) {
 
     private void createFastReadWriteCaches(Class clazz) {
         // The top level class might not be Compact serializable
-        CompactUtil.verifyClassIsCompactSerializable(clazz);
+        boolean isRegisteredAsCompact = compactStreamSerializer.isRegisteredAsCompact(clazz);
+        CompactUtil.verifyClassIsCompactSerializable(clazz, isRegisteredAsCompact);
```

Comment:
Here I believe `isRegisteredAsCompact` cannot be true. If it were, we would use that serializer